### ISSUE TITLE
Postprocessing the probabilities of an XGBoost Model

### ISSUE THREAD

**Issue by jfdatascience** [jpmml/sklearn2pmml#383]
Hello,

I have a question which i could not resolve with my tries and research.
Maybe someone kind and experienced knows what I am doing wrong.

My sklearn2pmml pipeline works without problems so far but I wanted to transform the output probabilities. They look like

```
       y  probability(0)  probability(1)
0      0        0.988244        0.011756
1      0        0.970045        0.029955
```

I simply wanted to multiply them by the factor of 1000 with the following adaption (also tried predict_proba_transformer)

```
pipeline_f_up = PMMLPipeline(steps=[('a', mapper),
                      ('b',  xgb_classifier_up)], predict_transformer= ExpressionTransformer("X[0] * 1000") )
```
                      
In the PMML file there seems to be no change.

```
					<Output>
						<OutputField name="probability(0)" optype="continuous" dataType="float" feature="probability" value="0"/>
						<OutputField name="probability(1)" optype="continuous" dataType="float" feature="probability" value="1"/>
					</Output>
```

thanks in advance

**Comment by vruusmann** [comment]
> Maybe someone kind and experienced knows what I am doing wrong.

What is your SkLearn2PMML package version?

Well, recently I've had a number of people come and complain that "things don't work as advertised", while using 3-5 year old JPMML software versions.

This issue looks exactly like the same.

> I wanted to transform the output probabilities.

You must set the `PMMLPipeline.predict_proba_transformer` attribute then.

Right now you're setting `predict_transformer` attribute, which is about the predicted label (ie. maps to the `XGBClassifier.predict(X)` method).

> In the PMML file there seems to be no change.

Fully reproducible demo case:

```python
from sklearn.datasets import load_iris
from sklearn2pmml import sklearn2pmml
from sklearn2pmml.pipeline import PMMLPipeline
from sklearn2pmml.preprocessing import ExpressionTransformer
from xgboost import XGBClassifier

iris_X, iris_y = load_iris(return_X_y = True)

pipeline = PMMLPipeline([
	("classifier", XGBClassifier(n_estimators = 3))
], predict_proba_transformer = ExpressionTransformer("X[0] * 1000"))
pipeline.fit(iris_X, iris_y)

sklearn2pmml(pipeline, "pipeline.pmml")
```

The generated PMML document contains the following markup:

```xml
<Output>
	<OutputField name="probability(0)" optype="continuous" dataType="float" feature="probability" value="0"/>
	<OutputField name="probability(1)" optype="continuous" dataType="float" feature="probability" value="1"/>
	<OutputField name="probability(2)" optype="continuous" dataType="float" feature="probability" value="2"/>
	<OutputField name="eval(X[0] * 1000)" optype="continuous" dataType="double" feature="transformedValue">
		<Apply function="*">
			<FieldRef field="probability(0)"/>
			<Constant dataType="integer">1000</Constant>
		</Apply>
	</OutputField>
</Output>
```

Works exactly as advertised.

**Comment by vruusmann** [comment]
> pipeline_f_up_pmml = make_pmml_pipeline(...)

The `sklearn2pmml.make_pmml_pipeline()` utility function does not take `predict_transformer`, `predict_proba_transformer` etc. extension attributes. The assumption is that this UF is being used to convert existing workflows, which don't have any post-processing logic attached to them. 

However, it's possible to set or unset these `PMMLPipeline` attributes manually:

``` python
pipeline_f_up_pmml = make_pmml_pipeline(...)
pipeline_f_up_pmml.predict_transformer = ExpressionTransformer(...)
pipeline_f_up_pmml.predict_proba_transformer = ExpressionTransformer(...)
```
